pygac/reader.py

```diff
@@ -205,19 +206,24 @@ def _correct_data_set_name(cls, header, filename):
             filename (str): path to file
         """
         filename = str(filename)
-        data_set_name = header['data_set_name'].decode(errors='ignore')
-        if not cls.data_set_pattern.match(data_set_name):
-            LOG.debug('The data_set_name in header %s does not match.'
-                      ' Use filename instead.' % header['data_set_name'])
+        for encoding in "utf-8", "cp500":
```

Comment:
The body of the for loop could be extracted to a separate method

Reply:
How would you go about the `break`? Should we raise an error in the submethod and catch that instead of breaking?

Reply:
Oh sorry, I overlooked the break... But yeah, that sounds good!